test/lit/metadce/sourcemap.wat

```diff
@@ -1,26 +1,34 @@
-;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
-;; RUN: wasm-metadce %s --graph-file %s.json -S -o - | filecheck %s
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-metadce %s --graph-file %s.json -S -o - | filecheck %s --check-prefix=TXT
 ;; RUN: wasm-as %s -o %t.wasm --source-map %t.map
 ;; RUN: wasm-metadce %t.wasm --input-source-map %t.map --graph-file %s.json -o %t.out.wasm --output-source-map %t.out.map
-;; RUN: wasm-dis %t.out.wasm --source-map %t.out.map -o - | filecheck %s
+;; RUN: wasm-dis %t.out.wasm --source-map %t.out.map -o - | filecheck %s --check-prefix=BIN
 
 ;; Test that sourcemap information is preserved
 
 (module
   ;;@ a:1:1
-  (func (export "f")
+  ;; TXT:      (type $0 (func))
+
+  ;; TXT:      (export "f" (func $f))
+
+  ;; TXT:      (func $f
+  ;; TXT-NEXT:  ;;@ a:2:1
+  ;; TXT-NEXT:  (nop)
+  ;; TXT-NEXT:  ;;@ a:3:1
+  ;; TXT-NEXT: )
+  (func $f (export "f")
      ;;@ a:2:1
      (nop)
      ;;@ a:3:1
   )
 )
-;; CHECK:      (type $0 (func))
+;; BIN:      (type $0 (func))
 
-;; CHECK:      (export "f" (func $0))
+;; BIN:      (export "f" (func $0))
 
-;; CHECK:      ;;@ a:1:1
```

Comment:
You are no longer checking that the prologue location is preserved.

Reply:
Yep, the test auto updater doesn't know about prologue source comments yet, so it's a trade off between having test coverage for them and making the test manual.